### PR TITLE
Remove jquery dependency from search_context.js

### PR DESCRIPTION
**Pull request by jcoyne**
Bootstrap 5 is removing the jquery dependency, so this gets Blacklight moving towards not requiring that dependency.

### CODE REVIEW

File: app/assets/javascripts/blacklight/blacklight.js [hunk 1]
```diff
@@ -432,54 +432,68 @@ Blacklight.onLoad(function () {
   Blacklight.modal.setupModal();
 });
 
-(function ($) {
-  Blacklight.doSearchContextBehavior = function () {
-    if (typeof Blacklight.do_search_context_behavior == 'function') {
-      console.warn("do_search_context_behavior is deprecated. Use doSearchContextBehavior instead.");
-      return Blacklight.do_search_context_behavior();
-    }
+Blacklight.doSearchContextBehavior = function () {
```
Comment (jcoyne) [inline app/assets/javascripts/blacklight/blacklight.js]: Note: this code is the compiled distribution. The source is in `app/javascript/`

File: app/javascript/blacklight/search_context.js [hunk 2]
```diff
@@ -1,49 +1,67 @@
...
+  document.body.appendChild(form)
+  form.querySelector('[type="submit"]').click()
+
+  event.preventDefault()
+  event.stopPropagation()
+  event.stopImmediatePropagation()
+};
+
+Blacklight.onLoad(function() {
+    Blacklight.doSearchContextBehavior();
```
Comment (mejackreed) [inline app/javascript/blacklight/search_context.js]:
```suggestion
  Blacklight.doSearchContextBehavior();
```

Comment (jcoyne) [reply app/javascript/blacklight/search_context.js]: Isn't this the job of jslint/jshint/hound?

Comment (jcoyne) [reply app/javascript/blacklight/search_context.js]: Seriously, why isn't hound catching this?  It's in the .jshint config.

Comment (mejackreed) [reply app/javascript/blacklight/search_context.js]: I'm thinking we should move away from jshint and hound and to an updated eslint config that just gets raised on CI. Hound / jshint hasn't been that reliable

**Pull request merged**